src/Discord.Net.Core/Entities/Messages/MessageReference.cs

```diff
@@ -39,16 +45,21 @@ public class MessageReference
         /// <param name="guildId">
         ///     The ID of the guild that will be referenced. It will be validated if sent.
         /// </param>
-        public MessageReference(ulong? messageId = null, ulong? channelId = null, ulong? guildId = null)
+        /// <param name="failIfNotExists">
+        ///     Whether to error if the referenced message doesn't exist instead of sending as a normal (non-reply) message. Defaults to true.
+        /// </param>
+        public MessageReference(ulong? messageId = null, ulong? channelId = null, ulong? guildId = null, bool? failIfNotExists = null)
         {
             MessageId = messageId ?? Optional.Create<ulong>();
             InternalChannelId = channelId ?? Optional.Create<ulong>();
             GuildId = guildId ?? Optional.Create<ulong>();
+            FailIfNotExists = failIfNotExists ?? Optional.Create<bool>();
```

Comment:
I'm open to swapping this to default to true (arguably a better behavior) -- left it this way to match the API.